File: src/Poker/game/ActionLog.java
package Poker.game;

import java.util.ArrayList;
import java.util.List;

public class ActionLog
{
   private List<ActionLogEntry> entries = new ArrayList<ActionLogEntry>();
   
   public void addEntry(Player player, String action)
   {
      addEntry(new ActionLogEntry(player, action));
   }
   
   public void addEntry(String action)
   {
      addEntry(new ActionLogEntry(action));
   }
   
   public void addEntry(ActionLogEntry entry)
   {
      entries.add(entry);
   }
   
   public void addEntries(List<ActionLogEntry> entries)
   {
      this.entries.addAll(entries);
   }
   
   public List<ActionLogEntry> getLog()
   {
      return entries;
   }
}

File: server/game/actionLog.js
let ActionLogEntry = require('./actionLogEntry');

let ActionLog = () =>
{
	let entries = [];
	
	let self =
	{
		addSystemEntry: (action) =>
		{
			self.addEntry(ActionLogEntry(action));
		},
		addPlayerEntry: (action, player) =>
		{
			self.addEntry(ActionLogEntry(action, [player]));
		},
		addEntry: (entry) =>
		{
			entries.push(entry);
		},
		addEntries: (newEntries) =>
		{
			entries = entries.concat(newEntries);
		},
		getEntries: () => entries
	}
	
	return self;
}

module.exports = ActionLog;
